snuba/query/parser/validator.py

```diff
@@ -5,12 +5,13 @@
 from snuba.query.logical import Query
 from snuba.query.parser.exceptions import ValidationException
 from snuba.query.validation import ExpressionValidator
+from snuba.query.validation.like import LikeFunctionValidator
 from snuba.state import get_config
 
 logger = logging.getLogger(__name__)
 
 default_validators: Sequence[ExpressionValidator] = [
-    # TODO: Add validators
+    LikeFunctionValidator(),
```

Comment:
Are you planning to add to this list a validator for every function (or group of ClickHouse functions) that we will be allowing then running our input through all of them? What about instead inspecting the function name and just picking the right validator that corresponds to that function rather than executing them all? We could immediately reject the ones with a function name that is not on our list.

Reply:
I don't plan to have a validator for all functions. At least not any time soon. In terms of list of validators I do not have a list as of now of what we should add. If we start adding more function parameters validators, yes definitively we would have one function validator with that inspects the name and applies the rules for that name.
I think I can refactor this this way in this PR stack. Will make the change in #1182 (this one is stacked on top of)

Reply:
Done, now the validators are in a mapping, and the validator is generic for any function signature.